namesys/republisher/repub_test.go

```diff
@@ -0,0 +1,120 @@
+package republisher_test
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	goprocess "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess"
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+
+	"github.com/ipfs/go-ipfs/core"
+	mock "github.com/ipfs/go-ipfs/core/mock"
+	namesys "github.com/ipfs/go-ipfs/namesys"
+	. "github.com/ipfs/go-ipfs/namesys/republisher"
+	mocknet "github.com/ipfs/go-ipfs/p2p/net/mock"
+	peer "github.com/ipfs/go-ipfs/p2p/peer"
+	path "github.com/ipfs/go-ipfs/path"
+)
+
+func TestRepublish(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// create network
+	mn := mocknet.New(ctx)
+
+	var nodes []*core.IpfsNode
+	for i := 0; i < 10; i++ {
+		nd, err := core.NewNode(ctx, &core.BuildCfg{
+			Online: true,
+			Host:   mock.MockHostOption(mn),
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		nodes = append(nodes, nd)
+	}
+
+	mn.LinkAll()
+
+	bsinf := core.BootstrapConfigWithPeers(
+		[]peer.PeerInfo{
+			nodes[0].Peerstore.PeerInfo(nodes[0].Identity),
+		},
+	)
+
+	for _, n := range nodes[1:] {
+		if err := n.Bootstrap(bsinf); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// have one node publish a record that is valid for 1 second
+	publisher := nodes[3]
+	p := path.FromString("/ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn") // does not need to be valid
+	rp := namesys.NewRoutingPublisher(publisher.Routing)
+	err := rp.PublishWithEOL(ctx, publisher.PrivateKey, p, time.Now().Add(time.Second))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	name := "/ipns/" + publisher.Identity.Pretty()
+	if err := verifyResolution(nodes, name, p); err != nil {
+		t.Fatal(err)
+	}
+
+	// Now wait a second, the records will be invalid and we should fail to resolve
+	time.Sleep(time.Second)
+	if err := verifyResolutionFails(nodes, name); err != nil {
+		t.Fatal(err)
+	}
+
+	// The republishers that are contained within the nodes have their timeout set
+	// to 12 hours. Instead of trying to tweak those, we're just going to pretend
+	// they dont exist and make our own.
```

Comment:
yeah, its good to test both: 
- that republishers work
- that republishers installed on the node work

the more tests the better, as it is we need way more coverage of all our expectations.

fine to merge for now, but we'll need something like this eventually. will be easier when we split things out